tests/console/yast2_vnc.pm

```diff
@@ -7,88 +7,87 @@
 # notice and this notice are preserved.  This file is offered as-is,
 # without any warranty.
 
-# Summary: Add test for yast2 vnc
+# Summary: Configure remote administration with yast2 vnc
 # Maintainer: Zaoliang Luo <zluo@suse.de>
 
 use strict;
 use base "console_yasttest";
 use testapi;
 use utils;
-use version_utils qw(is_leap is_sle);
+use version_utils qw(is_leap is_sle is_tumbleweed);
 use registration 'add_suseconnect_product';
 
-sub open_firewall_port {
-    # open port in firewall if it is enabled and check network interfaces, check long text by send key right.
-    assert_screen [qw(yast2_vnc_open_port_firewall yast2_vnc_firewall_port_closed)];
-    if (match_has_tag 'yast2_vnc_open_port_firewall' && (is_sle('<15') || is_leap('<15.0'))) {
-        send_key 'alt-p';
-        send_key 'alt-d';
-        assert_screen 'yast2_vnc_firewall_port_details';
-        send_key 'alt-e';
-        for (1 .. 5) { send_key 'right'; }
-        send_key 'alt-a';
-        send_key 'alt-o';
-    }
-    elsif (match_has_tag 'yast2_vnc_firewall_port_closed') {
-        send_key 'alt-f';    # Open port
-        assert_screen 'yast2_vnc_firewall_port_open';
-    }
-}
-
-sub run {
-    my $self = shift;
-    select_console 'root-console';
-
-    # check network at first
-    assert_script_run("if ! systemctl -q is-active network; then systemctl -q start network; fi");
-
-    # install components to test plus dependencies for checking
-    my $packages = 'vncmanager xorg-x11';
-    # netstat is deprecated in newer versions, use 'ss' instead
-    my $older_products = is_sle('<15') || is_leap('<15.0');
-    $packages .= ' net-tools' if $older_products;
-    if (check_var('ARCH', 's390x') && !$older_products) {
-        add_suseconnect_product('sle-module-desktop-applications', undef, undef, undef, 180);
-    }
-    zypper_call("in $packages");
-
-    # start Remote Administration configuration
-    script_run("yast2 remote; echo yast2-remote-status-\$? > /dev/$serialdev", 0);
+my $is_newer_product = (is_sle('>=15') || is_leap('>=15.0') || is_tumbleweed);
+my %remote_admin = (
```

Comment:
I general I like approach of moving shortcuts mapping to separate place, but why not to put them to the single hash? At least for me that would be easier. Name can include information for which module/dialog certain hotkey is relevant. E.g. `vnc_`, `fw_`. WDYT?

Reply:
We could create one hash and put inside the hashes related with every user control used in the module, but the call to that structure I found it a little cumbersome and might not worth it (at least...I though about it a time ago and now I don't remember very much but we could create some function that simplify the call to be nicer for the tester to call it)
In my opinion the prefix would be confusing, because anyway I would like to see the screen and when you see the screen and the mapping in hashes is direct. Not in this test, but it could exists things that are vnc_ and fw_ at the same time... What I mean it is that it could be difficult to label by ourself this thing or construct relationships among users controls under some tag/label, it is more direct to map control by control in my opinion.

Reply:
Hash of hashes is overkill, but for current approach, it's also more complex as when editing person has to know which hash to use and then the key, instead of just a key.
As for prefix, hash name is a prefix already. But in case we need hotkey for some generic button e.g. `Ok`, where to put it? In yet another hash? We also don't need prefixes when it's clear which button is for what.
As an alternative we could move all these mappings to separate file, so we never need to touch test code, but just shortkeys mapping and there we need a better structure than just single hash.

Reply:
Yes, I am still struggling with the best design :) your last suggestion is a good idea, I will move to another file and we can think more clear about it in future.

Reply:
We have discussed this topic with @OleksandrOrlov recently. I think we should follow up on that and agree on some more precise convention/guidelines at least inside of Y-subteam. In general we should get rid of most shortkeys by having new UI framework, but that's not going to happen next month.